### PR TITLE
#1283 Add streamApp development document

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1283)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: docs/custom_streamapp.md [hunk 1]
```diff
@@ -107,7 +107,51 @@ From now on, you can use the [StreamApp Java API](#streamapp-java-api) to design
...
+- OStream: define the functions for operating streaming data (each row record one-by-one)
+- OGroupedStream: define the functions for operating <b>grouped</b> streaming data
+- OTable: define the functions for operating table data (changelog for same key of row record)
+
+The above classes will be auto converted when you use the correspond functions; You should not worried about the usage 
+of which class is right to use. All you need nothing but **OStream**.
+
+Below we list the available functions in each classes (See more information in <i>javadoc</i>):
+
+##### OStream
```
Comment (chia7712) [inline docs/custom_streamapp.md]: Is "###" more suitable???

File: docs/custom_streamapp.md [hunk 2]
```diff
@@ -107,7 +107,51 @@ From now on, you can use the [StreamApp Java API](#streamapp-java-api) to design
...
+
+- constructTable(String topicName)
```
Comment (chia7712) [inline docs/custom_streamapp.md]: It would be better to briefly describe the method.

File: docs/custom_streamapp.md (hunk 1 above)
Comment (saivirtue) [reply docs/custom_streamapp.md]: Sure :)

File: docs/custom_streamapp.md (hunk 2 above)
Comment (saivirtue) [reply docs/custom_streamapp.md]: OK

File: docs/custom_streamapp.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
 # Custom StreamApp Guideline
 
-Ohara streamApp is a unparalleled wrap of [kafka streams](https://kafka.apache.org/documentation/streams) which gives
-a way to design your streaming flow.
+Ohara streamApp is an unparalleled wrap of [kafka streams](https://kafka.apache.org/documentation/streams) which gives
+you a straight thought to design your streaming flow.
```
[the remark this answers is not in the document]
Comment (saivirtue) [reply docs/custom_streamapp.md]: :+1 thanks

File: docs/custom_streamapp.md [hunk 4]
```diff
@@ -107,7 +107,51 @@ From now on, you can use the [StreamApp Java API](#streamapp-java-api) to design
...
 ## StreamApp Java API
 
-See the java doc in source code.
+In StreamApp, we provide three different classes for developers:
+- OStream: define the functions for operating streaming data (each row record one-by-one)
+- OGroupedStream: define the functions for operating <b>grouped</b> streaming data
+- OTable: define the functions for operating table data (changelog for same key of row record)
+
+The above classes will be auto converted when you use the correspond functions; You should not worried about the usage 
+of which class is right to use. All you need nothing but **OStream**.
```
[the remark this answers is not in the document]
Comment (saivirtue) [reply docs/custom_streamapp.md]: I change to `All the starting point of development...`, Is this more suitable?

[the remark this answers is not in the document]
Comment (konekoya) [reply docs/custom_streamapp.md]: Maybe _All you need is **OStream**_? Couldn't come up with something better, sorry 😅

**Comment by saivirtue** [comment]
retry connector

**Comment by saivirtue** [comment]
This pr only change document and javadoc, skip the unrelated fail tests

**Pull request merged**